Objects/typeobject.c

```diff
@@ -3506,30 +3501,30 @@ check_basicsize_includes_size_and_offsets(PyTypeObject* type)
                      "tp_basicsize for type '%s' (%d) is too small for base '%s' (%d)",
                      type->tp_name, type->tp_basicsize,
                      type->tp_base->tp_name, type->tp_base->tp_basicsize);
-        return 0;
+        return -1;
     }
     if (type->tp_weaklistoffset + (Py_ssize_t)sizeof(PyObject*) > max) {
         PyErr_Format(PyExc_TypeError,
                      "weaklist offset %d is out of bounds for type '%s' (tp_basicsize = %d)",
                      type->tp_weaklistoffset,
                      type->tp_name, type->tp_basicsize);
-        return 0;
+        return -1;
     }
     if (type->tp_dictoffset + (Py_ssize_t)sizeof(PyObject*) > max) {
         PyErr_Format(PyExc_TypeError,
                      "dict offset %d is out of bounds for type '%s' (tp_basicsize = %d)",
                      type->tp_dictoffset,
                      type->tp_name, type->tp_basicsize);
-        return 0;
+        return -1;
     }
     if (type->tp_vectorcall_offset + (Py_ssize_t)sizeof(vectorcallfunc*) > max) {
         PyErr_Format(PyExc_TypeError,
                      "vectorcall offset %d is out of bounds for type '%s' (tp_basicsize = %d)",
```

Comment:
The format is wrong: tp_vectorcall_offset and tp_basicsize types are Py_ssize_t: ``%zd`` should be used instead.